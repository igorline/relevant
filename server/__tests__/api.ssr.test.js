delete process.env.BROWSER;
process.env.API_SERVER = 'https://relevant.community';

const { _request } = require('utils/api');

jest.mock('../../server/api/user/user.controller', () => {
  return {
    test: jest.fn(params => params)
  };
});

const { test } = require('../../server/api/user/user.controller');

const getStore = () => ({
  community: {
    active: 'relevant'
  }
});

const options = {
  query: { search: 'user' },
  params: { _id: '123' },
  endpoint: 'user',
  path: 'test',
  method: 'POST',
  body: { data: 'some data' }
};

describe('server-side request', () => {
  it('generate correct fetch request', async () => {
    await _request(options, getStore);
    expect(test).toMatchSnapshot();
    expect(test).toBeCalled();
  });
});
