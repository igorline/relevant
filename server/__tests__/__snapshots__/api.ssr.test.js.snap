// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`server-side api request generate correct fetch request 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "body": Object {
          "data": "some data",
        },
        "params": Object {
          "_id": "123",
        },
        "query": Object {
          "community": "relevant",
          "search": "user",
        },
        "user": undefined,
      },
      null,
      [Function],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": Object {
        "body": Object {
          "data": "some data",
        },
        "params": Object {
          "_id": "123",
        },
        "query": Object {
          "community": "relevant",
          "search": "user",
        },
        "user": undefined,
      },
    },
  ],
}
`;
